runtime/int.go

```diff
@@ -430,6 +441,20 @@ func intCheckedMul(v, w int) (int, bool) {
 	return x, true
 }
 
+func intCheckedPow(v, w int) (int, bool) {
+	// first try the shortcut - assuming the this operation is
```

Comment:
On my system measuring a simple math.Exp vs. big.NewInt(0).Pow gives 4µs vs. 16µs. I have not yet tested the full code with all the surrounding functions.